blues/smartdart.py

```diff
@@ -5,55 +5,161 @@
 import simtk.unit as unit
 import numpy as np
 from blues.ncmc import SimNCMC
-def zero_masses( system, firstres, lastres):
-    for index in range(firstres, lastres):
-        system.setParticleMass(index, 0*daltons)
-
-def beta(temperature):
-    kB = unit.BOLTZMANN_CONSTANT_kB * unit.AVOGADRO_CONSTANT_NA
-    kT = kB * temperature
-    beta = 1.0 / kT
-    return beta
-
-
-
-def forcegroupify(system):
-    forcegroups = {}
-    for i in range(system.getNumForces()):
-        force = system.getForce(i)
-        force.setForceGroup(i)
-        forcegroups[force] = i
-    return forcegroups
-
-def getEnergyDecomposition(context, forcegroups):
-    energies = {}
-    for f, i in forcegroups.items():
-        energies[f] = context.getState(getEnergy=True, groups=2**i).getPotentialEnergy()
-    return energies
+import mdtraj as md
+
+#TODO consider throwing a warning if particle choices
+#are not dispersed enough (by some angle cutoff)
+def changeBasis(a, b):
+    '''
+    Changes positions of a particle (b) in the regular basis set to
+    another basis set (a).
+
+    Arguments
+    ---------
+    a: 3x3 np.array
+        Defines vectors that will create the new basis.
+    b: 1x3 np.array
+        Defines position of particle to be transformed into
+        new basis set.
+
+    Returns
+    -------
+    changed_coord: 1x3 np.array
+        Coordinates of b in new basis.
+
+    '''
+    ainv = np.linalg.inv(a.T)
+    print('ainv', ainv)
+    print('b.T', b.T)
+    changed_coord = np.dot(ainv,b.T)*unit.nanometers
+    return changed_coord
+def undoBasis(a, b):
+    '''
+    Transforms positions in a transformed basis (b) to the regular
+    basis set.
+
+    Arguments
+    ---------
+    a: 3x3 np.array
+        Defines vectors that defined the new basis.
+    b: 1x3 np.array
+        Defines position of particle to be transformed into
+        regular basis set.
+
+    Returns
+    -------
+    changed_coord: 1x3 np.array
+        Coordinates of b in new basis.
+    '''
+    a = a.T
+    print('a', a.T)
+    print('b.T', b.T)
+    changed_coord = np.dot(a,b.T)*unit.nanometers
+    return changed_coord
+
+
+def normalize(vector):
+    '''Normalize a given vector
+
+    Arguemnts
+    ---------
+    vector: 1xn np.array
+        Vector to be normalized.
+
+    Returns
+    -------
+    unit_vec: 1xn np.array
+        Normalized vector.
+    '''
+    magnitude = np.sqrt(np.sum(vector*vector))
+    unit_vec = vector / magnitude
+    return unit_vec
+def localcoord(particle1, particle2, particle3):
+    '''Defines a new coordinate system using 3 particles
+    returning the new basis set vectors
+    '''
+    part1 = particle1 - particle1
+    part2 = particle2 - particle1
+    part3 = particle3 - particle1
+#    vec1 = normalize(part2)
+#    vec2 = normalize(part3)
+    vec1 = part2
+    vec2= part3
+    vec3 = np.cross(vec1,vec2)*unit.nanometers
+    print('vec3', vec3, normalize(vec3))
```

Comment:
@nathanmlim @sgill2 - this will fail in Python2.7, yes? Since we're using the print function without importing it from `future`? 

@sgill2 - this code will need tests! (It's only passing right now because it doesn't have tests...)